tests/ui/codegen/duplicated-path-in-error.rs

```diff
@@ -1,5 +1,6 @@
 //@ only-linux
 //@ compile-flags: -Zcodegen-backend=/non-existing-one.so
+//@ normalize-stderr: "Error loading shared library .+:" -> "cannot open shared object file:"
```

Comment:
This depends on the specifics of the libc, but different libcs can have different specifics. This may just make a problem for another libc that wants our support.

Instead introduce the following:
```suggestion
//@ revisions: musl gnu
//@ only-linux
//@ compile-flags: -Zcodegen-backend=/non-existing-one.so
//@[gnu] only-gnu
//@[musl] only-musl
```

...also, add `"only-musl",` to right after here in src/tools/compiletest/src/directive-list.rs and I think it should work automatically:
https://github.com/rust-lang/rust/blob/c6a955468b025dbe3d1de3e8f3e30496d1fb7f40/src/tools/compiletest/src/directive-list.rs#L207